pkg/generate/code/set_resource.go

```diff
@@ -1229,6 +1238,29 @@ func SetResourceForStruct(
 	for memberIndex, memberName := range sourceShape.MemberNames() {
 		targetMemberShapeRef := targetShape.MemberRefs[memberName]
 		if targetMemberShapeRef == nil {
+			// This can happen when the targetShape is a primitive
+			// and the sourceShape is a struct like when EC2 resource DHCPOptions
+			// needs to set NewDhcpConfiguration.Values (targetShape = string) field from
+			// DhcpConfiguration.Values (sourceShape = AttributeValue). AttributeValue
+			// has a 'Value' member whereas the string shape does not. However,
+			// there is a possibility that the targetShape can be set from
+			// a different member within sourceShape using SetConfig.
+			if currentField, ok := r.Fields[currentPath]; ok {
+				setCfg := currentField.GetSetterConfig(model.OpTypeList)
```

Comment:
any particular reason for hardcoding "OpTypeList" here?

Reply:
nope just limited by api here and didn't want to introduce refactoring changes. Same as [here](https://github.com/aws-controllers-k8s/code-generator/blob/2eb18c5a67e5955149a6bb1eaa92440cdfed31e3/pkg/generate/ack/controller.go#L112)